polyfills/Map/polyfill.js

```diff
@@ -0,0 +1,124 @@
+(function(global) {
+
+
+	// Deleted map items mess with iterator pointers, so rather than removing them mark them as deleted. Can't use undefined or null since those both valid keys so use a private symbol.
+	var undefMarker = {};
+
+	// NaN cannot be found in an array using indexOf, so we encode NaNs using a private symbol.
+	var NaNMarker = {};
+
+	function encodeKey(key) {
+		return Number.isNaN(key) ? NaNMarker : key;
+	}
+	function decodeKey(encodedKey) {
+		return (encodedKey === NaNMarker) ? NaN : encodedKey;
+	}
+
+	function makeIterator(mapInst, getter) {
+		var nextIdx = 0;
+		var done = false;
+		return {
+			next: function() {
+				if (nextIdx === mapInst._keys.length) done = true;
+				if (!done) {
+					while (mapInst._keys[nextIdx] === undefMarker) nextIdx++;
+					return {value: getter.call(mapInst, nextIdx++), done: false};
+				} else {
+					return {done:true};
+				}
+			}
+		}
+	}
+
+	function calcSize(mapInst) {
+		var size = 0;
+		for (var i=0, s=mapInst._keys.length; i<s; i++) {
+			if (mapInst._keys[i] !== undefMarker) size++;
+		}
+		return size;
+	}
+
+	var ACCESSOR_SUPPORT = true;
+
+	var Map = function(data) {
+		this._keys = [];
+		this._values = [];
+
+		// If `data` is iterable (indicated by presence of a forEach method), pre-populate the map
+		data && data.forEach && data.forEach(function (item) {
+			this.set.apply(this, item);
+		}, this);
+
+		if (!ACCESSOR_SUPPORT) this.size = calcSize(this);
+	};
+	Map.prototype = {};
+
+	// Some old engines do not support ES5 getters/setters.  Since Map only requires these for the size property, we can fall back to setting the size property statically easch time the size of the map changes.
+	try {
+		Object.defineProperty(Map.prototype, 'size', {
+			get: function() {
+				return calcSize(this);
+			}
+		});
+	} catch(e) {
+		ACCESSOR_SUPPORT = false;
+	}
+
+	Map.prototype['get'] = function(key) {
```

Comment:
Is the array notation being used for consistency? IIRC `get` is not a reserved word and would be safe to use in dot-notation.